### PR TITLE
Fix batch and length normalization, add tests, replace distutils with setuptools

### PR DESCRIPTION
**Pull request by jpuigcerver**
In my previous PR there was a bug: the value of the normalized loss function was correct, but the gradients were those of the unnormalized version. This fixes that issue.

The user can select to normalize the loss (and gradients) w.r.t. the batch size (`size_average=True`) or w.r.t. the total number of frames in the batch (`length_average=True`). The latter overrides the former.

I updated the tests to prevent dummy mistakes like these.

In addition, Distutils support soon will be deprecated in pip, currently this warning is shown when one tries to uninstall the package with `pip uninstall`:

```
DEPRECATION: Uninstalling a distutils installed project (warpctc-pytorch) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
```

### CODE REVIEW

File: pytorch_binding/warpctc_pytorch/__init__.py [hunk 1]
```diff
@@ -36,24 +37,41 @@ def forward(ctx, acts, labels, act_lens, label_lens, size_average=False):
...
-            # Compute the avg. log-probability per frame and batch sample.
-            costs = torch.FloatTensor([costs.mean()])
-        else:
-            costs = torch.FloatTensor([costs.sum()])
+
+        costs = torch.FloatTensor([costs.sum()])
+
+        if length_average:
+            # Compute the avg. log-probability per batch sample and frame.
+            total_length = torch.prod(act_lens)
```
Comment (SeanNaren) [inline pytorch_binding/warpctc_pytorch/__init__.py]: Shouldn't this be sum instead of prod?

**Comment by SeanNaren** [comment]
Thanks man, I added a comment to the PR!

**Comment by jpuigcerver** [comment]
I can't find the comment...

**Comment by SeanNaren** [comment]
@jpuigcerver sorry forgot to publish the review...

File: pytorch_binding/warpctc_pytorch/__init__.py (hunk 1 above)
Comment (jpuigcerver) [reply pytorch_binding/warpctc_pytorch/__init__.py]: You are absolutely right. I didn't spot the error because in the tests `prod(act_lens) == sum(act_lens)` (i.e. `2 * 2 = 2 + 2`)...

**Comment by SeanNaren** [comment]
Thanks man :)
